eng/config/globalconfigs/ApiShim.globalconfig

```diff
@@ -2,11 +2,50 @@ is_global = true
 
 # API-related warnings do not make sense in API shim files
 
+# CA1707: Remove the underscores from member name 'name'
+dotnet_diagnostic.CA1707.severity = none
+
 # CA1816: Call GC.SuppressFinalize correctly
 dotnet_diagnostic.CA1816.severity = none
 
+# CA1822: Member 'Member' does not access instance data and can be marked as static
+dotnet_diagnostic.CA1822.severity = none
+
+# CA2227: Change 'Member' to be read-only by removing the property setter
+dotnet_diagnostic.CA2227.severity = none
+
 # IDE0044: Add readonly modifier
 dotnet_diagnostic.IDE0044.severity = none
 
 # IDE0060: Remove unused parameter
 dotnet_diagnostic.IDE0060.severity = none
+
+# RS0016: Add public types and members to the declared API
+dotnet_diagnostic.RS0016.severity = none
+
+# RS0026: Do not add multiple public overloads with optional parameters
+dotnet_diagnostic.RS0026.severity = none
+
+# RS0027: Public API with optional parameter(s) should have the most parameters amongst its public overloads
+dotnet_diagnostic.RS0027.severity = none
+
+# RS0037: Enable tracking of nullability of reference types in the declared API
+dotnet_diagnostic.RS0037.severity = none
+
+# RS0043: Do not call 'GetTestAccessor()'
+dotnet_diagnostic.RS0043.severity = none
+
+# VSTHRD002: Synchronously waiting on tasks or awaiters may cause deadlocks
+dotnet_diagnostic.VSTHRD002.severity = suggestion
```

Comment:
Was this firing in the shims? Ideally diagnostics which did not occur can be removed from this file, but otherwise set to `none`.

This comment certainly applies to the remaining VSTHRD and VSSDK diagnostics, but consider checking CA1707 above as well.